doc/buckets.md

```diff
@@ -1,122 +1,164 @@
+-->
 
-**NOTE:** For imports and thumbnails to work, the service account needs to create `signedURLs`. In order to do that, the service account must be granted with the role `roles/iam.serviceAccountTokenCreator` (or a custom role with `iam.serviceAccounts.signBlob`) inside the service account permissions (IAM > Service Accounts > `your_service_account` > Permissions).
+## Requirements
 
+- You need to create 3 buckets in your preferred Cloud provider:
+  - Import Bucket
+  - Client Bucket
+  - Thumbnails Bucket
 
-### Thumbnails
+> There's no name constraints
 
-```bash
-WORKSPACE_THUMBNAILS_PROVIDER='gcp'
-WORKSPACE_THUMBNAILS_BUCKET='bucket-name1'
-WORKSPACE_THUMBNAILS_KEYFILENAME='route_to_json'
-WORKSPACE_THUMBNAILS_PROJECTID='myproject'
-```
+> :warning: Map thumbnails can be configured in two different ways: public (map thumbnails storage objects are public) or private (map thumbnails storage objects are private). In order to control it, change the value of `appConfigValues.workspaceThumbnailsPublic` (boolean). Depending on this, the bucket properties (public access) may be different.
+
+- CORS configuration: Thumbnails and Import buckets require having the following CORS headers.
+  - Allowed origins: `*`
+  - Allowed methods: `GET`, `PUT`, `POST`
+  - Allowed headers (common): `Content-Type`, `Content-MD5`, `Content-Disposition`, `Cache-Control`
+    - GCS (extra): `x-goog-content-length-range`, `x-goog-meta-filename`
+    - Azure (extra): `Access-Control-Request-Headers`, `X-MS-Blob-Type`
+  - Max age: `3600`
+
+> CORS is configured at bucket level in GCS and S3, and at storage account level in Azure.
+
+> How do I setup CORS configuration? Check the provider docs: [GCS](https://cloud.google.com/storage/docs/configuring-cors), [AWS S3](https://docs.aws.amazon.com/AmazonS3/latest/userguide/enabling-cors-examples.html), [Azure Storage](https://docs.microsoft.com/en-us/rest/api/storageservices/cross-origin-resource-sharing--cors--support-for-the-azure-storage-services#enabling-cors-for-azure-storage).
+
+- Generate credentials to access those buckets, our supported authentication methods are:
+  - GCS: Service Account Key
+  - AWS: Access Key ID and Secret Access Key
+  - Azure: Access Key
+
+- Grant Read/Write permissions over the buckets to the credentials mentioned above.
+
+## Google Cloud Storage
+
+In order to use Google Cloud Storage custom buckets you need to:
+
+1. Create the buckets.
+   
+2. Configure the required CORS [settings](#requirements).
+   
+3. Create a [custom Service account](#custom-service-account).
+   
+4. Grant this service account with the following role (in addition to the buckets access): `roles/iam.serviceAccountTokenCreator`. 
 
+   > :warning: We don't recommend grating this role at project IAM level, but instead at the Service Account permissions level (IAM > Service Accounts > `your_service_account` > Permissions).
 
+   <!--
+   TODO: Add the code related to Terraform
```

Comment:
Instead of the code, just add a link to the proper `.tf` file in `carto-selfhosted/terraform/examples`